components/LayoutWithAuth0.js

```diff
@@ -45,60 +68,70 @@ class LayoutWithAuth0 extends Component {
     if (result.status === 'error') {
       window.alert(result.message)
     } else {
-      this.setState({apiUser: result})
+      // Refresh users.
+      await this.fetchUsers()
       // TODO: Push to a success page.
       router.push('/?newApiAccount=true')
     }
   }
 
-  _fetchUsers = async () => {
-    const { accessToken } = this.props.auth
-    // TODO: Combine into a single fetch fromToken or use SWR
-    const adminUserResult = await secureFetchHandleErrors(`${ADMIN_USER_URL}/fromtoken`, accessToken)
-    const apiUserResult = await secureFetchHandleErrors(`${API_USER_URL}/fromtoken`, accessToken)
-    this.setState({
-      adminUser: adminUserResult.data,
-      apiUser: apiUserResult.data,
-      isUserFetched: true
-    })
+  fetchUsers = async () => {
+    const { accessToken, isUserRequested } = this.state
+    if (!isUserRequested) {
+      // Set requested flag to avoid multiple requests, and void the previous user fetched state.
+      // TODO: useEffect?
+      this.setState({
+        isUserFetched: false,
+        isUserRequested: true
+      })
+
+      // TODO: Combine into a single fetch fromToken or use SWR
+      const adminUserResult = await secureFetchHandleErrors(`${ADMIN_USER_URL}/fromtoken`, accessToken)
+      const apiUserResult = await secureFetchHandleErrors(`${API_USER_URL}/fromtoken`, accessToken)
+      this.setState({
+        adminUser: adminUserResult.data,
+        apiUser: apiUserResult.data,
+        isUserFetched: true,
+        isUserRequested: false
+      })
+    }
   }
 
   updateUser = async ({user, type, isSelf}) => {
-    const { accessToken } = this.props.auth
+    const { accessToken } = this.state
     const result = await secureFetchHandleErrors(
       `${getUserUrl(type)}/${user.id}`,
       accessToken,
       'PUT',
       { body: JSON.stringify(user) }
     )
     // Ensure user object gets updated at top level if updating self.
-    if (isSelf) this._fetchUsers()
+    if (isSelf) this.fetchUsers()
     if (result.status === 'error') {
       window.alert(result.message)
     }
   }
 
   async componentDidUpdate () {
-    const { auth } = this.props
-    const { accessToken, isAuthenticated } = auth
-    const { isUserRequested } = this.state
+    const { auth0 } = this.props
 
-    if (isAuthenticated && accessToken && !isUserRequested) {
-      // Set a flag to prevent duplicate fetches while awaiting the calls below to return.
-      this.setState({ isUserRequested: true })
+    if (this.acccessTokenIsUnfetched()) {
+      const accessToken = await auth0.getAccessTokenSilently()
+      this.setState({ accessToken })
```

Comment:
Were you able to verify if setting the access token in the state was necessary? I'm wondering whether the auth0 library also uses react state hooks to store it if already fetched.

Reply:
@landonreed Could you clarify? We are passing the token value around to `secureFetch` in the render method of the component, and it is going to be tricky to fit `async getAccessTokenSilently` in the render method.

Reply:
How we're handling right now is probably OK, but I think we should add a FIXME about using something a little more contained for securing our API calls with the access token. Maybe pointing to this or another example: https://github.com/auth0/auth0-react/blob/master/EXAMPLES.md#4-create-a-useapi-hook-for-accessing-protected-apis-with-an-access-token

Reply:
Noted in https://github.com/ibi-group/otp-admin-ui/pull/30/commits/d01a8a1c5e2894325ca8d5de21f1c729996ebd16.

Reply:
Further changes to access token management are now in #33, which removes `accessToken` from the state of the component.